Integration Testing CI Test Failure on python 3.12

The daily CI failed on test for scanpy failed.  Please go to [the logs of the integration testing repo](https://github.com/scverse/integration-testing/actions/runs/16209637574) to review. @scverse/anndata

Is legacy-api-wrap broken? 

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/testing/scanpy/_helpers/data.py:47: in krumsiek11
    return _krumsiek11().copy()
           ^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/scanpy/datasets/_datasets.py:163: in krumsiek11
    adata = read(HERE / "krumsiek11.txt", first_column_names=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/legacy_api_wrap/__init__.py:82: in fn_compatible
    return fn(*args_all, **kw)
           ^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/scanpy/readwrite.py:151: in read
    return _read(
/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/scanpy/readwrite.py:879: in _read
    adata = read_text(filename, delimiter, first_column_names)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args_all = (PosixPath('/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/scanpy/datasets/krumsiek11.txt'), None, True)
kw = {}
args_pos = (PosixPath('/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/scanpy/datasets/krumsiek11.txt'), None)
args_rest = (True,)

    @wraps(fn)
    def fn_compatible(*args_all: P.args, **kw: P.kwargs) -> R:
        if len(args_all) <= n_positional:
            return fn(*args_all, **kw)
    
        args_pos: P.args
        args_pos, args_rest = args_all[:n_positional], args_all[n_positional:]
    
        if len(args_rest) > len(old_positionals):
            n_max = n_positional + len(old_positionals)
            msg = (
                f"{fn.__name__}() takes from {n_required} to {n_max} parameters, "
                f"but {len(args_pos) + len(args_rest)} were given."
            )
            raise TypeError(msg)
>       warn(
            f"The specified parameters {old_positionals[:len(args_rest)]!r} are "
            "no longer positional. "
            f"Please specify them like `{old_positionals[0]}={args_rest[0]!r}`",
            category=category,
            stacklevel=stacklevel,
        )
E       FutureWarning: The specified parameters ('first_column_names',) are no longer positional. Please specify them like `first_column_names=True`

/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/legacy_api_wrap/__init__.py:94: FutureWarning
```
This seems to be a non-positional argument...

--- a/src/scanpy/readwrite.py
+++ b/src/scanpy/readwrite.py
@@ -876,7 +876,7 @@ def _read(  # noqa: PLR0912, PLR0915
                 "... assuming '.data' means tab or white-space separated text file"
             )
             logg.hint("change this by passing `ext` to sc.read")
-        adata = read_text(filename, delimiter, first_column_names)
+        adata = read_text(filename, delimiter, first_column_names=first_column_names)
     elif ext == "soft.gz":
         adata = _read_softgz(filename)
     elif ext == "loom":

